akka-persistence-typed/src/main/scala/akka/persistence/typed/javadsl/EventSourcedOnCommandBehavior.scala

```diff
@@ -0,0 +1,260 @@
+package akka.persistence.typed.javadsl
+
+import akka.actor.typed
+import akka.actor.typed.BackoffSupervisorStrategy
+import akka.actor.typed.Behavior
+import akka.actor.typed.internal.BehaviorImpl.DeferredBehavior
+import akka.actor.typed.javadsl.ActorContext
+import akka.annotation.InternalApi
+import akka.persistence.typed.EventAdapter
+import akka.persistence.typed.NoOpEventAdapter
+import akka.persistence.typed.PersistenceId
+import akka.persistence.typed.SnapshotAdapter
+import akka.persistence.typed.SnapshotSelectionCriteria
+import akka.persistence.typed.internal
+import akka.persistence.typed.internal.EffectImpl
+import akka.persistence.typed.internal.NoOpSnapshotAdapter
+import akka.persistence.typed.scaladsl
+import akka.util.unused
+
+import java.util.Collections
+import java.util.Optional
+
+/**
+ * Behavior
+ * For Java versions before 17, see [[EventSourcedBehavior]]
+ */
+abstract class EventSourcedOnCommandBehavior[Command, Event, State](
```

Comment:
do we need one for EventSourcedBehaviorWithEnforcedReplies too?

Reply:
Yep, soon in a new push.